galaxy_importer/collection.py

```diff
@@ -57,8 +58,20 @@ def _import_collection(file, filename, logger, cfg):
     with tempfile.TemporaryDirectory() as tmp_dir:
         sub_path = 'ansible_collections/placeholder_namespace/placeholder_name'
         extract_dir = os.path.join(tmp_dir, sub_path)
-        with tarfile.open(fileobj=file, mode='r') as pkg_tar:
-            pkg_tar.extractall(extract_dir)
+        os.makedirs(extract_dir)
+
+        cmd = [
+            'tar',
+            f'--directory={extract_dir}',
+            '-xf',
+            file.name,
+        ]
+        proc = Popen(cmd, cwd=PULP_ARTIFACT_DIR)
```

Comment:
You could simply use `subprocess.run()` to run command and return its exit code.

Reply:
Updated